lib/attache/upload.rb

```diff
@@ -12,10 +12,26 @@ def _call(env, config)
         params   = request.params
         return config.unauthorized unless config.authorized?(params)
 
-        relpath = generate_relpath(Attache::Upload.sanitize params['file'])
-        cachekey = File.join(request_hostname(env), relpath)
+        if params.has_key? 'data'
+
+          base_64_encoded_data = params['data']
+          data_index = base_64_encoded_data.index('base64') + 7
+          data_format = base_64_encoded_data.match(/image\/(\w+)/)[1]
+          filedata = base_64_encoded_data[data_index..-1]
+          decoded_image = Base64.decode64(filedata)
+
+          filename = "#{Time.now.to_i}.#{data_format}"
```

Comment:
Can we use the value of `params[:file]` for filename instead, as per existing api?

I'm open to have a fallback, auto generated filename but if we do, it should apply for both base64 or binary file upload